Content.Server/GameObjects/Components/Power/PowerNetComponents/BatteryDischargerComponent.cs

```diff
@@ -31,14 +33,14 @@ public override void ExposeData(ObjectSerializer serializer)
         public override void Initialize()
         {
             base.Initialize();
-
-            _battery = Owner.EnsureComponent<BatteryComponent>();
-            _supplier = Owner.EnsureComponent<PowerSupplierComponent>();
```

Comment:
be careful, **componentdependencies do NOT ensure components**, they only tryget it. imo this is faulty logic

maybe we should add a `ensure` flag to componentdependencies :thinking: